more_itertools/more.py

```diff
@@ -3447,3 +3448,40 @@ def windowed_complete(iterable, n):
         middle = seq[i : i + n]
         end = seq[i + n :]
         yield beginning, middle, end
+
+
+def all_unique(iterable, key=None):
+    """
+    Returns ``True`` if all the elements of *iterable* are unique (no two
+    elements are equal).
+
+    If a *key* function is specified, it will be used to preprocess the
+    elements of the iterable before comparing.
+
+        >>> all_unique('ABCB')
+        False
+        >>> all_unique('ABCb')
+        True
+        >>> all_unique('ABCb', str.lower)
+        False
+
+    Sequences with a mix of hashable and unhashable items can be used, but the
+    function will be slower for unhashable items. This is similar to
+    :func:`unique_everseen`, see details there.
+    """
+    seenset = set()
+    seenset_add = seenset.add
+    seenlist = []
+    seenlist_add = seenlist.append
+    for element in map(key, iterable) if key else iterable:
+        try:
```

Comment:
Do we want to use `if isinstance(element, Hashable)` instead of the `try`/`except`?
Would require importing `Hashable` from `collections.abc`.

Reply:
I copied the implementation from `unique_everseen`. Do we maybe want to ensure that it works the same way? If not I don't mind changing it to the hashable check

Reply:
Hmm. Maybe the suggested changes make more sense for subsequent PR that changes the implementation of both functions?